curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64caeb134c3cdc5498cd75b9.md

```diff
@@ -0,0 +1,351 @@
+---
+id: 64caeb134c3cdc5498cd75b9
+title: Step 84
+challengeType: 0
+dashedName: step-84
+---
+
+# --description--
+
+Inside that array, add a conditional statement that checks if the player's `x` position is greater than or equal to the platform's `x` position minus half of the player's width.
+
+# --hints--
+
+You should have a conditional statement that checks if the player's `x` position is greater than or equal to the platform's `x` position minus half of the player's width.
+
+```js
+assert.match(code, /const\s*platformDetectionRules\s+=\s*\[\s*player\.position\.x\s*>=\s*platform\.position\.x\s*-\s*player\.width\s*\/\s*2,?\s*\];?/)
+
```

Comment:
This is the hint and test I was trying to add in:

This is the hint and test I was trying to add in:

You should have a conditional statement that checks if the player's `x` position is greater than or equal to the platform's `x` position minus half of the player's width.

```js
assert.match(code, /const\s*platformDetectionRules\s+=\s*\[\s*player\.position\.x\s*>=\s*platform\.position\.x\s*-\s*player\.width\s*\/\s*2,?\s*\];?/)
```